docs/reference/query-dsl/knn-query.asciidoc

```diff
@@ -275,3 +278,9 @@ Thus, the final results from aggregations contain
 `k * number_of_shards` documents. This is different from
 the <<knn-search,top level knn section>> where aggregations are
 calculated on the global top `k` nearest documents.
+
+[[knn-query-multiple-indices]]
+==== Knn query over multiple indices
+`knn` query calculates the top `k` documents from each index searched.
+Therefore, if `k` is smaller than the query `size`, `k` matching documents
+from each index will be returned.
```

Comment:
This is not correct, `k` is per shard in the knn query so `k` results per shard will be evaluated. The number of documents to return is determined by the `size` parameter and has nothing to do with `k`.

Reply:
Corrected, but added the clarification so users won't be confused. Let me know what you think.

Reply:
What are we trying to explain here? This is still not correct (`k` is per shard and not per index) and add confusion imo. What is confusing for users, I still don't get it?

Reply:
Removed.